test/e2e/gardener/shoot/create_migrate_delete.go

```diff
@@ -5,90 +5,101 @@
-		f.Shoot = shoot
-
+	test := func(s *ShootContext) {
 		// Assign seedName so that shoot does not get scheduled to the seed that will be used as target.
-		f.Shoot.Spec.SeedName = ptr.To(getSeedName(false))
-
-		It("Create, Migrate and Delete", Offset(1), func() {
-			By("Create Shoot")
-			ctx, cancel := context.WithTimeout(parentCtx, 30*time.Minute)
-			defer cancel()
-
-			Expect(f.CreateShootAndWaitForCreation(ctx, false)).To(Succeed())
-			f.Verify()
-
-			// TODO: add back VerifyInClusterAccessToAPIServer once this test has been refactored to ordered containers
-			// if !v1beta1helper.IsWorkerless(s.Shoot) && !v1beta1helper.HibernationIsEnabled(s.Shoot) {
-			// 	// We can only verify in-cluster access to the API server before the migration in local e2e tests.
-			// 	// After the migration, the shoot API server's hostname still points to the source seed, because
-			// 	// the /etc/hosts entry is never updated. Hence, we talk to the API server for starting in-cluster
-			// 	// clients. That's also why the ShootMigrationTest is configured to skip all interactions with the
-			// 	// shoot API server for local e2e tests.
-			// 	inclusterclient.VerifyInClusterAccessToAPIServer(s)
-			// }
-
-			By("Migrate Shoot")
-			ctx, cancel = context.WithTimeout(parentCtx, 20*time.Minute)
-			defer cancel()
-			t, err := newDefaultShootMigrationTest(ctx, f.Shoot, f.GardenerFramework)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(t.MigrateShoot(ctx)).To(Succeed())
-			Expect(t.VerifyMigration(ctx)).To(Succeed())
+		s.Shoot.Spec.SeedName = ptr.To(getSeedName(false))
+
+		ItShouldCreateShoot(s)
+		ItShouldWaitForShootToBeReconciledAndHealthy(s)
+		ItShouldInitializeShootClient(s)
+		ItShouldGetResponsibleSeed(s)
+		ItShouldInitializeSeedClient(s)
+
+		if !v1beta1helper.IsWorkerless(s.Shoot) && !v1beta1helper.HibernationIsEnabled(s.Shoot) {
+			// We can only verify in-cluster access to the API server before the migration in local e2e tests.
+			// After the migration, the shoot API server's hostname still points to the source seed, because
+			// the /etc/hosts entry is never updated. Hence, we talk to the API server for starting in-cluster
+			// clients. That's also why the ShootMigrationTest is configured to skip all interactions with the
+			// shoot API server for local e2e tests.
+			inclusterclient.VerifyInClusterAccessToAPIServer(s)
+		}
 
-			By("Delete Shoot")
-			ctx, cancel = context.WithTimeout(parentCtx, 20*time.Minute)
-			defer cancel()
-			Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
+		var (
+			seedClientSourceCluster client.Client
+			secretsBeforeMigration  map[string]corev1.Secret
+		)
+
+		It("Record current seed client", func() {
+			seedClientSourceCluster = s.SeedClient
 		})
+
+		It("Populate comparison elements before migration", func(ctx SpecContext) {
+			var err error
+			secretsBeforeMigration, err = GetPersistedSecrets(ctx, s.SeedClientSet, s.Shoot.Status.TechnicalID)
```

Comment:
Can you wrap client calls like `GetPersistedSecrets` in an `Eventually` block? This should make the e2e tests more stable by retrying.